Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM python:3.9-bullseye
+
+ENV PIP_NO_CACHE_DIR=1
+RUN apt-get update && apt-get -y upgrade && \
+    apt-get -y install gdal-bin python3-gdal && \
+    apt-get -y autoremove && \
+    apt-get clean
+
+COPY . /app
```

Comment:
It should be better to do 

```
RUN mkdir /app
COPY requirements.docker.txt /app/requirements.docker.txt
COPY setup.py /app/setup.py
```
then COPY the other useful files after the RUN of the 3 pip commands.

By this way, the python install is done only once and Docker uses the cache (except if requirements.docker.txt or setup.py change) instead of doing it again at each edit on the repository.

In addition by copying only needed files, we have a smaller image.

Reply:
that's a genuine suggestion , Once pip package is published i will not need whole app directory .  In order to install package for now i need to rely on the repo . Do you have better idea to install package for now ?

Reply:
I am not sure to understand your question.

My suggestion is for development setup: put the copy of files which are edited at the end to use at the maximum Docker cache. To run the pip commands with minimal file copy, we only need requirements.docker.txt and setup.py. By copying only these 2 files before running pip commands, Docker will cache all the pip setup until we edit these files. So if we update the API without updating the python dependencies, Docker won't built again the depencies.

For deployment, you can install the galaxy-api package with pip install and GitHub url ([example](https://pythoninoffice.com/python-pip-install-from-github/)).
But like we have now Docker, we can simply use the built Docker image by publishing on DockerHub for example. I see that hotosm has already [an account on DockerHub](https://hub.docker.com/u/hotosm).
Building and publishing a Docker image on DockerHub (or other Docker registry) can be automated with GitHub actions.

Reply:
oh ! that makes it clear , You are talking about cache of python dependencies since they will not get often changed , Included in [2b1c0b1](https://github.com/hotosm/galaxy-api/pull/335/commits/2b1c0b1d8ab46f7b6e0e959117d3e30661943058)